### PR TITLE
workflow-test

### PR DESCRIPTION
**Pull request by ulfgebhardt**
@darkdragon-001 Wanna take a look at this? I have no clue of python

Related: https://docs.github.com/en/actions/guides/building-and-testing-python

### CODE REVIEW

**Comment by ulfgebhardt** [comment]
image

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  ##############################################################################
+  # JOB: Test ##################################################################
+  ##############################################################################
+  test:
+    name: test
+    runs-on: ubuntu-latest
+    #needs: [nothing]
+    strategy:
+      matrix:
+        python-version: [2.7, 3.5, 3.6, 3.7, 3.8]
```
Comment (JBBgameich) [inline .github/workflows/test.yml:16]: We are no longer supporting python 2.7, and given it's long deprecated and unsupported itself, we should not invest work into that IMO.

Comment (darkdragon-001) [reply .github/workflows/test.yml:16]: f-strings are supported from 3.6 onwards. All major distros support that, so I guess its fine.

Comment (ulfgebhardt) [reply .github/workflows/test.yml]: I removed 2.7 and 3.5 from the list. Can you guys help me get this running so we have some automated testing?
Furthermore this is the basis to get the crawler running on the github servers as a cron and extract data regularly and fully automatically

**Comment by ulfgebhardt** [comment]
image

**Comment by JBBgameich** [comment]
@ulfgebhardt This looks like Python 2 syntax. I think you need to rebase your branch to get the Python 3 porting commits that are now in master.

**Comment by JBBgameich** [comment]
Apart from the python 2 syntax errors this looks good already. You can probably remove the pytest step for now as we do not have any tests that it could run.

We can add the functionality to run the scripts and extract laws in an additional pull request.
It would probably be good to work on one script that runs all the scrapers and converters we have in the correct order first. That would also be easy to test locally. After that works we can simply run that on GitHub actions.

**Comment by ulfgebhardt** [comment]
image

Could you lint this for me and push to this branch? @JBBgameich

**Comment by JBBgameich** [comment]
Not sure what you mean, fixing some of this warnings would mean a larger refactoring.
Do you want me to fix the warnings or to adjust the lint parameters to only show the more important instead of the stylistic warnings?

**Comment by ulfgebhardt** [comment]
What ever you please - I believe enforcing a code style is wise - but you might disagree. My aim is to get this into master so we have an "objective" reason to merge or not merge certain changes. Furthermore it would allow us to extend the existing tests in the future and have more detailed test going.

**Comment by JBBgameich** [comment]
I agree on the coding style part, but the linter complains about some functions being too complex, and that is not a thing that is very easy to fix

**Comment by ulfgebhardt** [comment]
As soon as you are satisfied please approve it yourself and merge. That way we have the check in the master and can enforce it before merge

**Comment by JBBgameich** [comment]
The only warning left is "E722 do not use bare 'except'".
Exactly for the reason using an except without naming the exception it is bad, I can't fix it right now, as I don't know which exceptions can happen.

**Review by JBBgameich — APPROVED** [review]

**Comment by ulfgebhardt** [comment]
Thanks alot <3

image

I now enforce the checks, so please rebase

**Pull request merged**